### PR TITLE
feat(*.model.ts): complete initial schemas

### PR DESCRIPTION
**Pull request by shivamvijaywargi**

### CODE REVIEW

File: src/models/user.model.ts
Comment (sub1120) [inline src/models/user.model.ts:1]: Can move IUser Interface to src/types?

Comment (shivamvijaywargi) [reply src/models/user.model.ts:1]: I think it is good to keep it in the model file for mongoose models at least, as I have generally seen people doing the same. We can still move it anytime in the future if required.

Comment (sub1120) [inline src/models/user.model.ts:1]: I think socialProfiles type, needs to modified a bit, like below
```
{
   type:string          // linkedin, twitter, insta
   profileUrl: string
}
```

What do you say?

Comment (sub1120) [inline src/models/user.model.ts:1]: for `passwordChangedAt` we can use `Date` Interface instead of `number`.

Comment (shivamvijaywargi) [reply src/models/user.model.ts:1]: Yeah, we can do that. But then we'll have to add many 'cause fb, insta, YT may not work in China. So we have to take care of such things and add many providers which we might not have even heard of. What do you think?

But yeah, adding them looks like a good idea to me as well.

Comment (shivamvijaywargi) [reply src/models/user.model.ts:1]: Ah yes, my bad. Looks like I need to go to bed now😅

Comment (sub1120) [reply src/models/user.model.ts:1]: For all other providers, we can have "Other" type. 

For the popular onces we can show them with their icons, if we already have profile type. (eg. for linkedIn type, we can use linkedIn logo in user profile)

Comment (sub1120) [reply src/models/user.model.ts:1]: Okay, let keep it now there. Will see in future, if required.

Comment (shivamvijaywargi) [reply src/models/user.model.ts:1]: Array of objects?

Alright we will discuss it in detail tomorrow.

Comment (sub1120) [inline src/models/user.model.ts:1]: Do we need `await` before `jwt.sign()` statement? I think it returns string, not promise

File: src/utils/currency.ts
Comment (sub1120) [inline src/utils/currency.ts:1]: Nice!!

File: src/models/user.model.ts
Comment (sub1120) [reply src/models/user.model.ts:1]: Sure

File: src/models/user.model.ts [hunk 1]
```diff
@@ -13,6 +13,13 @@ export interface IUser extends Document {
   password: string;
   passwordChangedAt: number;
   role: string;
+  biography: string;
+  socialProfiles: string[];
+  profilePicture: string;
+  slug: string;
+  rating: number;
+  totalStudents: number;
+  enrolledCourses: Schema.Types.ObjectId[];
```
Comment (sub1120) [inline src/models/user.model.ts:22]: So, are we going with reference or embed?

File: src/models/user.model.ts [hunk 2]
```diff
@@ -75,6 +82,41 @@ const userSchema = new Schema<IUser>(
...
+      type: Number,
+      min: [1, "Rating must be atleast 1"],
+      max: [5, "Rating must be less than or equal to 5"],
+      // default: 0,
+    },
+    totalStudents: {
+      type: Number,
+      default: 0,
+    },
+    enrolledCourses: [
```
Comment (sub1120) [inline src/models/user.model.ts:114]: So finally are we going with reference for enrolledCourses?

Comment (shivamvijaywargi) [reply src/models/user.model.ts:114]: It'll be updated, this was prior to when we had a discussion about it.

File: src/models/user.model.ts
Comment (shivamvijaywargi) [reply src/models/user.model.ts:1]: Maybe, I might have had issues in the past so that is why I await for it.

Comment (shivamvijaywargi) [reply src/models/user.model.ts:1]: Done

**Review by shivamvijaywargi — COMMENTED** [review]
Resolved current conversations

**Review by sub1120 — APPROVED** [review]
All good.

**Pull request merged**